conans/client/require_resolver.py

```diff
@@ -68,19 +68,40 @@ def resolve(self, require, base_conanref):
             raise ConanException("The version in '%s' from requirement '%s' could not be resolved"
                                  % (version_range, require))
 
+    @staticmethod
+    def _filter_latest_revisions(local_found_refs, search_ref):
```

Comment:
With the search results for a reference, we take the latest revision and then apply the version range resolution.